### ISSUE TITLE
v4.0.0 missing static binaries in Release page

### ISSUE THREAD

**Issue by nunix** [containers/podman#13298]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

The static binaries are missing from the `Release` page, however it's mentioned in the Readme they should be available.

**Steps to reproduce the issue:**

1. Only 2 assets are on the Release page (instead of 7 for 3.4.4)

**Describe the results you received:**
Only source code is available to download.

**Describe the results you expected:**
The static binaries for Linux/MacOS/Windows should be listed

**Comment by mheon** [comment]
I was told not to build them yet by @baude - I'll defer explanation as to why to him

**Comment by nunix** [comment]
@mheon thanks a lot for your prompt reply (on a Sunday 😄 ). It's OK if there's short delay, however the Readme should maybe not mention that the static binaries are available then (IMO)

**Comment by baude** [comment]
@nunix The reason why is they are unusable at this point unless the backend is Linux.  As with podman 3, podman3 remote clients must connect to podman3 backends.  Podman4 must connect with Podman4 backends.  Given that FCOS does not ship Podman4 yet, the mac client is not much help (unless connecting to you your Linux backend).  You could also compile the client as well.  They can be cross-compiled if you like.

Bottom line, we should have a better answer mid-werrk.

**Comment by nunix** [comment]
@baude thank you for the detailed answer, totally understandable. I will compile the `msi` as I did for the RC4 as I want to showcase the new `podman machine` with the WSL2 backend.

Again, my point was more the misalignment between the repo Readme stating the static binaries were already available and not finding them on release page. But sincerely all good, this case for me can be closed, or kept open just to avoid duplicates (if any) until the binaries are available.

Thanks again, and have a great rest of Sunday 😄

**Comment by afbjorklund** [comment]
When I run the release version, the client says "4.0.0-dev" and the server says "4.0.0-rc5"

```
Client:       Podman Engine
Version:      4.0.0-dev
API Version:  4.0.0-dev
```

```
Server:       Podman Engine
Version:      4.0.0-rc5
API Version:  4.0.0-rc5
```

But it doesn't seem to be using any podman3 backend, even if it is based on Fedora 35 ?

**Comment by afbjorklund** [comment]
I think it accidentally tagged the commit before...

**Comment by mheon** [comment]
Uh-oh. You are entirely correct.

I can recreate the tag from the correct commit, but that could confuse things?

**Comment by afbjorklund** [comment]
It happens...

I guess it will be fixed in 4.0.1, and it doesn't have much to do with the static `podman-remote` binaries.

**Comment by afbjorklund** [comment]
@mheon this also meant that Podman 4.0.0 still uses the 3.3.1 API (not 4.0.0 API)

**Comment by mheon** [comment]
Yep, we realized that yesterday. Current plan is to cut a 4.0.1 tomorrow, to give a bit of time for critical issues to emerge (if they're going to).

**Comment by gonzalocasas** [comment]
Perhaps it is obvious, but is not only in the readme. The download link for Windows [on the `Getting started` page of the website](https://podman.io/getting-started/installation) also [ends up in 404](https://github.com/containers/podman/releases/download/v4.0.0/podman-remote-release-windows.zip).

**Comment by fjorgemota** [comment]
> I don't think a new issue for "v4.0.1 is also missing static binaries in Release page" would help ?

I don't think so, but I think it honestly would help if this was clearly announced both on the release notes and on the README of the project, just to set expectations correctly. and avoid these sorts of issues being opened..

**Comment by afbjorklund** [comment]
I think static builds for `podman-remote` (client) were _supposed_ to be there, but not for `podman` (server).

**Comment by fjorgemota** [comment]
> I think static builds for podman-remote (client) were supposed to be there, but not for podman (server).

Nope, like @baude explained in a comment on [this same issue here](https://github.com/containers/podman/issues/13298#issuecomment-1046292593), podman-remote (the client) depends on the podman (server), and FCOS (Fedora CoreOS, the distro that's used on the virtual machine that podman-remote manages) wasn't updated yet to v4.0.

IMHO what could be released is the podman (server) versions for every distro, but at least on Ubuntu-side like #13302 explains the Kubic repository will not be maintained anymore, so...yeah. It's a complicated situation.

**Comment by afbjorklund** [comment]
I understand that it is some kind of weird side-track, but I did think that it was "released" as in "out there".

It (`v4.0.1`) seemed OK on Linux, downloaded fedora-coreos-35.20220225.dev.0-qemu.x86_64.qcow2.xz

```console
$ podman-remote version
Client:       Podman Engine
Version:      4.0.1
API Version:  4.0.1
Go Version:   go1.17.2
Git Commit:   c8b9a2e3ec3630e9172499e15205c11b823c8107
Built:        Wed Feb 23 20:34:57 2022
OS/Arch:      linux/amd64

Server:       Podman Engine
Version:      4.0.1
API Version:  4.0.1
Go Version:   go1.16.14

Built:      Fri Feb 25 18:05:11 2022
OS/Arch:    linux/amd64
$ podman-remote machine ssh rpm -q podman
Warning: Permanently added '[localhost]:45253' (ECDSA) to the list of known hosts.
podman-4.0.1-1.fc35.x86_64
```

Waiting for Podman Desktop 4.1.0 is probably best, if there is something else with these early builds.

**Comment by afbjorklund** [comment]
Tried rawhide, but it was still older.

`podman-4.0.0-0.6.rc4.fc37.x86_64`

**Comment by afbjorklund** [comment]
> the distro that's used on the virtual machine that podman-remote manages

it is `podman machine` that manages CoreOS, `podman-remote` can talk to any OS

**Comment by farfromunique** [comment]
(This may be the wrong place for this, but it's at least related)
How can I build the Podman-remote binary for windows? I'm not at all familiar with compiled-language software development...

**Comment by afbjorklund** [comment]
> How can I build the Podman-remote binary for windows?

Normally it is `make podman-remote-windows`, but please open a new issue if it doesn't work using the instructions

https://podman.io/getting-started/installation#building-from-scratch

**Comment by farfromunique** [comment]
Following that link (and searching for "Windows") tells me to install the latest remote client or read a blog post about how to install it for WSL. It says nothing about the `make` command.
It may be wise to update the documentation.
![image](https://user-images.githubusercontent.com/4977323/156410438-cac5fe06-edac-4d09-aa3c-e8092dbe4712.png)

**Comment by afbjorklund** [comment]
That assumes that there _is_ a release, which might be try again for 4.1 or so

**Comment by baude** [comment]
Things will get updated today for our 4.0.2 release.

**Comment by farfromunique** [comment]
@baude That was SO easy!  
For anyone who finds this later, here's what the instructions probably should be for v4.0.2 on Windows, with WSL2 backend:
```
download and run podman-v4.0.2.msi (it doesn't look like it does anything, but it does; just very fast!)
Powershell:
1) podman machine init (wait for it to be done. it'll tell you to run...)
2) podman machine start (wait ...)
3) podman run hello-world
```

**Comment by afbjorklund** [comment]
There are static podman-remote binaries for v4.0.2 now.

https://github.com/containers/podman/releases/tag/v4.0.2

**Comment by nunix** [comment]
Thanks a lot for the follow-up 😄

**Comment by gonzalocasas** [comment]
sorry to insist, but the main entry point for most people still returns an error when trying to download the binary:

![image](https://user-images.githubusercontent.com/933277/156636962-93b25bc0-a296-4392-85e2-180e6116a927.png)

**Comment by rhatdan** [comment]
@gonzalocasas Try it now.

**Comment by gonzalocasas** [comment]
yay! thanks, it works now!
